routes/dashboards.js

```diff
@@ -0,0 +1,272 @@
+var express = require('express');
+var router = express.Router();
+let moment = require('moment');
+const { check, validationResult } = require('express-validator');
+const uuidv4 = require('uuid/v4');
+var body = require('express-validator'); //validation
+const { Op } = require('sequelize');
+
+var ROLES = require('../utils/roles');
+var fs = require('fs');
+const fetch = (...args) => import('node-fetch').then(({ default: fetch }) => fetch(...args));
+
+var initModels = require('../models/init-models');
+var sequelise = require('../config/db/db_sequelise');
+const user = require('../models/user');
+
+var models = initModels(sequelise);
+
+//Initial Render of Admin Dashboard
+router.get(
+  '/admin',
+  require('connect-ensure-login').ensureLoggedIn({ redirectTo: '/app/auth/login' }),
+  function (req, res, next) {
+    if (req.user.role === ROLES.Admin) {
+      models.FoodprintHarvest.findAll({
+        order: [['pk', 'DESC']],
+      })
+        .then(harvest_rows => {
+          for (let i = 0; i < harvest_rows.length; i++) {
+            harvest_rows[i].harvest_photoHash =
+              'data:image/png;base64,' +
+              new Buffer(harvest_rows[i].harvest_photoHash, 'binary').toString('base64');
+          }
+          models.FoodprintStorage.findAll({
+            order: [['pk', 'DESC']],
+          }).then(storage_rows => {
+            res.render('dashboard_admin', {
+              title: 'FoodPrint - Admin Dashboard',
+              harvest_data: harvest_rows,
+              storage_data: storage_rows,
+              user: req.user,
+              page_name: 'Dashboard',
+            });
+          });
+        })
+        .catch(err => {
+          req.flash('error', err);
+          res.render('error', {
+            message: 'Unexpected Error occured',
+            data: '',
+            user: req.user,
+            page_name: 'error',
+          });
+        });
+    } else {
+      res.render('error', {
+        message: 'You are not authorised to view this resource.',
+        title: 'Error',
+        user: req.user,
+        page_name: 'error',
+      });
+    }
+  }
+);
+
+//Render of Admin Dashboard with filtering
+router.get(
+  '/admin/filter/:range',
+  require('connect-ensure-login').ensureLoggedIn({ redirectTo: '/app/auth/login' }),
+  function (req, res, next) {
+    if (req.user.role === ROLES.Admin) {
+      //Dates
+      let start_date = 0;
```

Comment:
@DanielmAaron does this not break if an unexpected filter is entered into the URL bar by a user e.g. 10-year. I'd suggest adding a final else that defaults to 1 month and perhaps a console log that an unexpected filter was received. 

Also, I'd suggest adding another variable like `current_date =  moment(new Date())` so that you are not repeatedly calling  `moment(new Date())`. This would be followed by `let start_date = current_date` so that you are not initialising `start_date` to zero (as it is not an integer, an alternative would be `let start_date = null`). Then finally you will have  `start_date =  current_date.subtract('1', 'months').format('YYYY-MM-DD HH:mm:ss');`